gcloud-java-compute/src/test/java/com/google/gcloud/compute/RawImageConfigurationTest.java

```diff
@@ -0,0 +1,86 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import com.google.gcloud.compute.ImageConfiguration.SourceType;
+
+import org.junit.Test;
+
+public class RawImageConfigurationTest {
+
+  private static final String SOURCE = "source";
+  private static final SourceType SOURCE_TYPE = SourceType.RAW;
+  private static RawImageConfiguration CONFIGURATION = RawImageConfiguration.builder(SOURCE)
+      .sourceType(SOURCE_TYPE)
+      .build();
+
+  @Test
+  public void testToBuilder() {
+    compareRawImageConfiguration(CONFIGURATION, CONFIGURATION.toBuilder().build());
+    String newSource = "newSource";
+    RawImageConfiguration configuration = CONFIGURATION.toBuilder().source(newSource).build();
+    assertEquals(newSource, configuration.source());
+    configuration = configuration.toBuilder().source(SOURCE).build();
+    compareRawImageConfiguration(CONFIGURATION, configuration);
+  }
+
+  @Test
+  public void testToBuilderIncomplete() {
+    RawImageConfiguration configuration = RawImageConfiguration.of(SOURCE);
+    compareRawImageConfiguration(configuration, configuration.toBuilder().build());
+  }
+
+  @Test
+  public void testBuilder() {
+    assertEquals(SOURCE_TYPE, CONFIGURATION.sourceType());
+    assertEquals(SOURCE, CONFIGURATION.source());
+  }
+
+  @Test
+  public void testToAndFromPb() {
+    assertTrue(ImageConfiguration.fromPb(CONFIGURATION.toPb()) instanceof RawImageConfiguration);
+    compareRawImageConfiguration(CONFIGURATION,
+        ImageConfiguration.<RawImageConfiguration>fromPb(CONFIGURATION.toPb()));
+    RawImageConfiguration configuration = RawImageConfiguration.of(SOURCE);
+    compareRawImageConfiguration(configuration, RawImageConfiguration.fromPb(configuration.toPb()));
+  }
+
+  @Test
+  public void testOf() {
+    RawImageConfiguration configuration = RawImageConfiguration.of(SOURCE);
+    assertNull(configuration.sourceType());
+    assertEquals(SOURCE, configuration.source());
+  }
+
+  @Test
+  public void testSetProjectId() {
+    assertSame(CONFIGURATION, CONFIGURATION.setProjectId("project"));
+  }
+
+  private void compareRawImageConfiguration(RawImageConfiguration expected,
+      RawImageConfiguration value) {
+    assertEquals(expected, value);
+    assertEquals(expected.source(), value.source());
+    assertEquals(expected.sourceType(), value.sourceType());
```

Comment:
Could also assert things specific to a `RawImageConfiguration`, like containerType, sha1Checksum, and archiveSizeBytes.